CompileAST cannot use imported packages

### The following program `sample.go` triggers an unexpected result

```go
func TestIssueX(t *testing.T) {
	const src = `
			package main

			func main() {
				fmt.Println("Hello")
			}
		`

	fset := token.NewFileSet()
	i := interp.New(interp.Options{})
	err := i.Use(stdlib.Symbols)
	if err != nil {
		t.Fatal(err)
	}
	_, err = i.Eval(`import "fmt"`)
	if err != nil {
		t.Fatal(err)
	}

	ast, err := parser.ParseFile(fset, "_.go", src, parser.DeclarationErrors)
	if err != nil {
		t.Fatal(err)
	}
	prog, err := i.CompileAST(ast)
	if err != nil {
		t.Fatal(err)
	}
	_, err = i.Execute(prog)
	if err != nil {
		t.Fatal(err)
	}
}
```


### Expected result

```console
$ go test
=== RUN   TestIssueX
Hello
--- PASS: TestIssueX (0.00s)
PASS
ok      github.com/traefik/yaegi/interp 0.007s
```


### Got

```console
$ go test
=== RUN   TestIssueX
    /home/firelizzard/src/yaegi/interp/interp_eval_test.go:1781: 1:1: undefined: fmt 17
--- FAIL: TestIssueX (0.00s)
FAIL
FAIL    github.com/traefik/yaegi/interp 0.007s
```


### Yaegi Version

5665c9a4108e5f878c0d1d8d1e1913eb3c314154

### Additional Notes

_No response_

--- a/interp/compile_test.go
+++ b/interp/compile_test.go
@@ -10,7 +10,8 @@ import (
 )
 
 func TestCompileAST(t *testing.T) {
-	file, err := parser.ParseFile(token.NewFileSet(), "_.go", `
+	i := New(Options{})
+	file, err := parser.ParseFile(i.FileSet(), "_.go", `
 		package main
 
 		import "fmt"
@@ -61,7 +62,6 @@ func TestCompileAST(t *testing.T) {
 		{desc: "expr", node: dFunc.Body.List[0]},
 	}
 
-	i := New(Options{})
 	_ = i.Use(stdlib.Symbols)
 
 	for _, c := range cases {

--- a/interp/interp_eval_test.go
+++ b/interp/interp_eval_test.go
@@ -5,6 +5,7 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"go/parser"
 	"io"
 	"log"
 	"net/http"
@@ -1747,5 +1748,38 @@ func TestRestrictedEnv(t *testing.T) {
 	})
 	if s, ok := os.LookupEnv("foo"); ok {
 		t.Fatal("expected \"\", got " + s)
+	}
+}
+
+func TestIssue1383(t *testing.T) {
+	const src = `
+			package main
+
+			func main() {
+				fmt.Println("Hello")
+			}
+		`
+
+	interp := interp.New(interp.Options{})
+	err := interp.Use(stdlib.Symbols)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = interp.Eval(`import "fmt"`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ast, err := parser.ParseFile(interp.FileSet(), "_.go", src, parser.DeclarationErrors)
+	if err != nil {
+		t.Fatal(err)
+	}
+	prog, err := interp.CompileAST(ast)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = interp.Execute(prog)
+	if err != nil {
+		t.Fatal(err)
 	}
 }

--- a/interp/program.go
+++ b/interp/program.go
@@ -3,6 +3,7 @@ package interp
 import (
 	"context"
 	"go/ast"
+	"go/token"
 	"os"
 	"reflect"
 	"runtime"
@@ -14,6 +15,12 @@ type Program struct {
 	pkgName string
 	root    *node
 	init    []*node
+}
+
+// FileSet is the fileset that must be used for parsing Go that will be passed
+// to interp.CompileAST().
+func (interp *Interpreter) FileSet() *token.FileSet {
+	return interp.fset
 }
 
 // Compile parses and compiles a Go code represented as a string.
@@ -55,6 +62,9 @@ func (interp *Interpreter) compileSrc(src, name string, inc bool) (*Program, err
 // CompileAST builds a Program for the given Go code AST. Files and block
 // statements can be compiled, as can most expressions. Var declaration nodes
 // cannot be compiled.
+//
+// WARNING: The node must have been parsed using interp.FileSet(). Results are
+// unpredictable otherwise.
 func (interp *Interpreter) CompileAST(n ast.Node) (*Program, error) {
 	// Convert AST.
 	pkgName, root, err := interp.ast(n)

